Hi @mertilginoglu , I reviewed your PR but I faced another issues when I try to test it in my local:

```
2026-02-15 12:57:15    FastAPI   Starting production server 🚀
2026-02-15 12:57:15  
2026-02-15 12:57:15              Searching for package file structure from directories with         
2026-02-15 12:57:15              __init__.py files                                                  
2026-02-15 12:57:15 INFO:         Logger initialized
2026-02-15 12:57:15              Importing from /code
2026-02-15 12:57:15  
2026-02-15 12:57:15     module   📁 app            
2026-02-15 12:57:15              ├── 🐍 __init__.py
2026-02-15 12:57:15              └── 🐍 main.py    
2026-02-15 12:57:15  
2026-02-15 12:57:15       code   Importing the FastAPI app object from the module with the following
2026-02-15 12:57:15              code:                                                              
2026-02-15 12:57:15  
2026-02-15 12:57:15              from app.main import app
2026-02-15 12:57:15  
2026-02-15 12:57:15        app   Using import string: app.main:app
2026-02-15 12:57:15  
2026-02-15 12:57:15     server   Server started at http://0.0.0.0:4200
2026-02-15 12:57:15     server   Documentation at http://0.0.0.0:4200/docs
2026-02-15 12:57:15  
2026-02-15 12:57:15              Logs:
2026-02-15 12:57:15  
2026-02-15 12:57:15 INFO:         Connected to NATS at nats-server:4222
2026-02-15 12:57:15 
2026-02-15 12:57:15       INFO   Started server process [1]
2026-02-15 12:57:15       INFO   Waiting for application startup.
2026-02-15 12:57:15      ERROR   Traceback (most recent call last):                                 
2026-02-15 12:57:15                File                                                             
2026-02-15 12:57:15              "/usr/local/lib/python3.13/site-packages/starlette/routing.py",    
2026-02-15 12:57:15              line 692, in lifespan                                              
2026-02-15 12:57:15                  async with self.lifespan_context(app) as maybe_state:          
2026-02-15 12:57:15                             ~~~~~~~~~~~~~~~~~~~~~^^^^^                          
2026-02-15 12:57:15                File "/usr/local/lib/python3.13/contextlib.py", line 214, in     
2026-02-15 12:57:15              __aenter__                                                         
2026-02-15 12:57:15                  return await anext(self.gen)                                   
2026-02-15 12:57:15                         ^^^^^^^^^^^^^^^^^^^^^                                   
2026-02-15 12:57:15                File "/code/app/main.py", line 18, in lifespan                   
2026-02-15 12:57:15                  await nats_client.js.add_stream(                               
2026-02-15 12:57:15                  ...<3 lines>...                                                
2026-02-15 12:57:15                  )                                                              
2026-02-15 12:57:15                File                                                             
2026-02-15 12:57:15              "/usr/local/lib/python3.13/site-packages/nats/js/manager.py", line 
2026-02-15 12:57:15              118, in add_stream                                                 
2026-02-15 12:57:15                  resp = await self._api_request(                                
2026-02-15 12:57:15                         ^^^^^^^^^^^^^^^^^^^^^^^^                                
2026-02-15 12:57:15                  ...<3 lines>...                                                
2026-02-15 12:57:15                  )                                                              
2026-02-15 12:57:15                  ^                                                              
2026-02-15 12:57:15                File                                                             
2026-02-15 12:57:15              "/usr/local/lib/python3.13/site-packages/nats/js/manager.py", line 
2026-02-15 12:57:15              421, in _api_request                                               
2026-02-15 12:57:15                  raise APIError.from_error(resp["error"])                       
2026-02-15 12:57:15                        ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^                       
2026-02-15 12:57:15                File "/usr/local/lib/python3.13/site-packages/nats/js/errors.py",
2026-02-15 12:57:15              line 89, in from_error                                             
2026-02-15 12:57:15                  raise BadRequestError(**err)                                   
2026-02-15 12:57:15              nats.js.errors.BadRequestError: nats: BadRequestError: code=400    
2026-02-15 12:57:15              err_code=10058 description='stream name already in use with a      
2026-02-15 12:57:15              different configuration'
2026-02-15 12:57:15      ERROR   Application startup failed. Exiting.
```

--- a/server/webhook-listener/app/main.py
+++ b/server/webhook-listener/app/main.py
@@ -4,7 +4,9 @@ from contextlib import asynccontextmanager
 from fastapi import Body, FastAPI, HTTPException, Header, Request, status
 from pydantic import BaseModel
 from nats.js.api import StreamConfig
+from nats.js.errors import BadRequestError
 from app.config import settings
+from app.logger import logger
 from app.nats_client import nats_client
 
 RETENTION_DAYS = 28
@@ -12,19 +14,27 @@ SECONDS_PER_DAY = 24 * 60 * 60
 RETENTION_MAX_AGE_SECONDS = RETENTION_DAYS * SECONDS_PER_DAY
 
 
+async def ensure_stream(name: str, subjects: list[str]) -> None:
+    stream_config = StreamConfig(
+        name=name,
+        subjects=subjects,
+        storage="file",
+        max_age=RETENTION_MAX_AGE_SECONDS,
+    )
+    try:
+        await nats_client.js.add_stream(config=stream_config)
+    except BadRequestError as error:
+        if getattr(error, "err_code", None) != 10058:
+            raise
+        await nats_client.js.update_stream(config=stream_config)
+        logger.info(f"Updated existing stream configuration for '{name}'")
+
+
 @asynccontextmanager
 async def lifespan(app: FastAPI):
     await nats_client.connect()
-    await nats_client.js.add_stream(
-        name="github",
-        subjects=["github.>"],
-        config=StreamConfig(storage="file", max_age=RETENTION_MAX_AGE_SECONDS),
-    )
-    await nats_client.js.add_stream(
-        name="notification",
-        subjects=["notification.>"],
-        config=StreamConfig(storage="file", max_age=RETENTION_MAX_AGE_SECONDS),
-    )
+    await ensure_stream(name="github", subjects=["github.>"])
+    await ensure_stream(name="notification", subjects=["notification.>"])
     yield
     await nats_client.close()
 

